applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -555,30 +548,28 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateOnIntegrationPoints(const
 
         RetentionLaw::Parameters RetentionParameters(this->GetProperties());
 
-        for (unsigned int GPoint = 0; GPoint < number_of_integration_points; ++GPoint) {
+        for (unsigned int integration_point = 0; integration_point < number_of_integration_points;
+             ++integration_point) {
             // Compute Np, GradNpT, B and StrainVector
-            this->CalculateKinematics(Variables, GPoint);
+            this->CalculateKinematics(Variables, integration_point);
 
             RetentionParameters.SetFluidPressure(GeoTransportEquationUtilities::CalculateFluidPressure(
                 Variables.Np, Variables.PressureVector));
-            rOutput[GPoint] = mRetentionLawVector[GPoint]->CalculateValue(
-                RetentionParameters, rVariable, rOutput[GPoint]);
+            rOutput[integration_point] = mRetentionLawVector[integration_point]->CalculateValue(
+                RetentionParameters, rVariable, rOutput[integration_point]);
         }
     } else if (rVariable == HYDRAULIC_HEAD) {
-        const PropertiesType& rProp = this->GetProperties();
-
         // Defining the shape functions, the Jacobian and the shape functions local gradients containers
-        const Matrix& NContainer = rGeom.ShapeFunctionsValues(mThisIntegrationMethod);
+        const Matrix& n_container = r_geometry.ShapeFunctionsValues(mThisIntegrationMethod);
 
-        const auto NodalHydraulicHead =
-            GeoElementUtilities::CalculateNodalHydraulicHeadFromWaterPressures(rGeom, rProp);
+        const auto nodal_hydraulic_head = GeoElementUtilities::CalculateNodalHydraulicHeadFromWaterPressures(
+            r_geometry, this->GetProperties());
 
-        for (unsigned int GPoint = 0; GPoint < number_of_integration_points; ++GPoint) {
-            double HydraulicHead = 0.0;
-            for (unsigned int node = 0; node < TNumNodes; ++node)
-                HydraulicHead += NContainer(GPoint, node) * NodalHydraulicHead[node];
-
-            rOutput[GPoint] = HydraulicHead;
+        for (unsigned int integration_point = 0; integration_point < number_of_integration_points;
+             ++integration_point) {
+            rOutput[integration_point] =
+                MathUtils<double>::Dot(row(n_container, integration_point), nodal_hydraulic_head);
```

Comment:
I'm glad that you are reusing existing functionality. In this case, however, I would suggest to use `std::inner_product` rather than `MathUtils<double>::Dot`. The reason for this suggestion is that the latter requires an implicit conversion from the return type of `row` to `Vector`, which involves heap allocation for the temporary object. The former uses iterators and therefore, there are no implicit conversions.
```suggestion
            const auto& shape_function_values = row(n_container, integration_point);
            rOutput[integration_point] =
                std::inner_product(shape_function_values.begin(), shape_function_values.end(),
                                   nodal_hydraulic_head.begin(), 0.0);
```

Reply:
done